src/librustc_data_structures/graph_algorithms/iterate/mod.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use super::Graph;
+use super::node_vec::NodeVec;
+
+#[cfg(test)]
+mod test;
+
+pub fn post_order_from<G: Graph>(graph: &G, start_node: G::Node) -> Vec<G::Node> {
```

Comment:
we should look at the existing iterators for MIR and remove the duplication. In these versions I opted to just generate a vector, which is way simpler -- it'd be interesting to check if the anyone is taking advantage of the "lazy" nature of the existing iterators.